### ISSUE TITLE
Extra Utilities Loading Crash

### ISSUE THREAD

**Issue by CadeEvs** [rwtema/extrautilities_old_issues#1005]
Hello, so I was working on a modpack for some friends and I and when I saw Extra Utilities updated I check the changelog and was really happy, I updated it and ran the game and during loading it crashed. At first I thought It had the error from Refined Storages Item Grid because I changed the recipe with CraftTweaker so I removed the script and It didnt fix the problem.

http://pastebin.com/fL0TZJDV

**Comment by b0bst3r** [comment]
Looks like a clash with the cofh API which Ex Utils ships, the cofh APIs in Ext Utils and Ref Stor are different.

**Comment by CadeEvs** [comment]
Hm, I removed the mod Refined Storage and it started loading fine

**Comment by b0bst3r** [comment]
Yeah both mods ship COFH API which are different and clashing, removing one or the other mod will obv stop the APIs from clashing and "fix" the issue.

Since Extra Utils is loaded first it's most likely loading that version of COFH API which doesn't have something Refined Storage needs, hence why Ref Storage is crashing out.

**Comment by CadeEvs** [comment]
Just downloaded them both on a separate profile and it loaded.

**Comment by b0bst3r** [comment]
Hmmm maybe it's another mod's cofh API that's causing it @raoulvdberge

**Comment by raoulvdberge** [comment]
RS uses the COFH api class `cofh/api/energy/ItemEnergyContainer`, and it seems to be missing in your installation. That means some other mod is messing with the COFH api, and not RS.

**Comment by CadeEvs** [comment]
I just loadedout EnderIO and it worked

**Comment by CadeEvs** [comment]
yea, It was EnderIO

**Comment by b0bst3r** [comment]
Which is a bit odd as EIO doesn't ship the COFH API

**Comment by b0bst3r** [comment]
@CadevsGaming Can I suggest you back EnderIO up to build 116 as I'm not having this issue with that build and just try it?

**Comment by CadeEvs** [comment]
Yes, it worked. :)

**Comment by NoobTaco** [comment]
It's not Ender IO. . I just launched with Ender IO, Ender Core and Ender Zoo disabled and it still crashes.

https://gist.github.com/NoobTaco/00ff049d5049322c2e1b87b7145adb88

**Comment by CadeEvs** [comment]
Well it has to do with the 117 Update

**Comment by b0bst3r** [comment]
@NoobTaco that's a completely different crash to do with tinkers construct :)

**Comment by NoobTaco** [comment]
@b0bst3r woot! More integration bugs!! The hunt continues!  :)

**Comment by b0bst3r** [comment]
EnderCore-1.10.2-0.4.1.56-beta
EnderIO-1.10.2-3.0.1.117_beta
extrautils2-1.10.2-alpha-1.1.0a
refinedstorage-1.1.2
jei_1.10.2-3.12.5.303
MCMultiPart-1.2.1-universal
CodeChickenLib-1.10.2-2.2.3.73-universal

Crashes with just those mods present.

**Comment by HenryLoenwind** [comment]
If that original stack trace wasn't cut off half-way through there would be a diagnostics section that lists where each RF class is loaded from.

PS: Don't fiddle around with installed mods for API loading problems. That will change the load order, hiding the error if the bad mod isn't the first one to be loaded anymore. Completely misleading and not helpful.

**Comment by b0bst3r** [comment]
@CadevsGaming you can close this, turns out it was EnderIO's issue which has been fixed.

**Comment by HenryLoenwind** [comment]
For everyone wondering: Make extra sure to never ever indirectly classload any class that's not in your jar (or part of Minecraft/Forge) when your main mod class or any of your proxy classes are loaded/instantiated. Unless that mod as been classloaded before your mod by pure luck, this will blacklist that class from ever being loaded.
